src/lib.rs

```diff
@@ -286,30 +281,33 @@ impl Default for RequestParams {
 }
 
 /// Get a default `Client` and `RequestParams`.
-pub fn default() -> (hyper::Client, RequestParams) {
-    (hyper::Client::new(), RequestParams::default())
+pub fn default() -> (reqwest::Client, RequestParams) {
+    //FIXME: Bad in libs
+    let client = reqwest::Client::new().unwrap();
+    (client, RequestParams::default())
 }
 
 macro_rules! req_with_body {
     ($client:ident, $url:ident, $req:ident, $params:ident, $method:ident) => ({
     	let body = $req.body.expect("Expected this request to have a body. This is a bug, please file an issue on GitHub.");
 
 		let body: &[u8] = (*body).borrow();
-		let mut cursor = Cursor::new(body);
+//FIXME: My current understanding of copying, etc. doesn't get me far enough to grok this
```

Comment:
Forget most of what I just said! We can use zero-copy readers in `reqwest` using the `Body::new` method.

So if you change the request line to:

```rust
$client.request(reqwest::Method::$method, &$url).headers($params.headers.to_owned()).body(Body::new(Cursor::new(body))).send()
```

Then all should be good.